src/generate-type-defs.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+const distDir = __dirname + "/../dist/";
+const specPath = require.resolve('shift-spec-idl/spec.idl');
+const attributeOrderPath = require.resolve('shift-spec-idl/attribute-order.conf');
+
+const fs = require('fs');
+const specConsumer = require('shift-spec-consumer');
+
+const specContents = fs.readFileSync(specPath, 'utf8');
+const attributeOrderContents = fs.readFileSync(attributeOrderPath, 'utf8');
+const {nodes, enums, namedTypes} = specConsumer(specContents, attributeOrderContents);
+
+function printType(type) {
+  switch (type.kind) {
+    case 'type':
+      return `Const(TYPE_INDICATOR), value: "${type.argument}"`;
+    case 'nullable':
+      return printType(type.argument) + ' | null | void';
```

Comment:
Is there a reason for this to be `null | void`, instead of just `null`? `void` doesn't mean the same thing in TypeScript as Flow, if I understand correctly, and is kind of weird here in either case. I wouldn't want these properties to be undefined or missing; I'd prefer they be required to be explicitly `null`.